Dockerfile

```diff
@@ -1,55 +1,34 @@
-FROM python:3.8-slim
+FROM python:3.8-slim as base
 
+FROM base as builder
 # Any python libraries that require system libraries to be installed will likely
 # need the following packages in order to build
 RUN apt-get update && apt-get install -y build-essential git
 
-RUN pip install pipenv
-ENV PIPENV_NOSPIN=true
-ENV PIPENV_HIDE_EMOJIS=true
 ENV CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
 
 ARG install_dev_dependencies=true
 
-RUN mkdir -p /app
+WORKDIR /app
 
 # Install stac_fastapi.types
-COPY ./stac_fastapi_types /app/stac_fastapi_types
-WORKDIR /app/stac_fastapi_types
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
-
-# Install stac_api.extensions.
-COPY ./stac_fastapi_extensions /app/stac_fastapi_extensions
-WORKDIR /app/stac_fastapi_extensions
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
-
-# Install stac_api.api.
-COPY ./stac_fastapi_api /app/stac_fastapi_api
-WORKDIR /app/stac_fastapi_api
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
-
-# Install stac_api.sqlalchemy.
-COPY ./stac_fastapi_sqlalchemy /app/stac_fastapi_sqlalchemy
-WORKDIR /app/stac_fastapi_sqlalchemy
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
-
-# Install stac_api.server.
-COPY ./stac_fastapi_server /app/stac_fastapi_server
-WORKDIR /app/stac_fastapi_server
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
-
-# Install base package
-COPY ./setup.py /app/
-COPY ./Pipfile /app/
-COPY ./Pipfile.lock /app/
-WORKDIR /app
-RUN pipenv install --deploy --ignore-pipfile ${install_dev_dependencies:+--dev}
+COPY . /app
+
+ENV PATH=$PATH:/install/bin
 
+RUN mkdir -p /install && \
+    pip install -e ./stac_fastapi/api[dev]   && \
+    pip install -e ./stac_fastapi/types[dev]  && \
+    pip install -e ./stac_fastapi/extensions[dev,tiles]  && \
+    pip install -e ./stac_fastapi/sqlalchemy[dev,server]
```

Comment:
Why use `-e` in a docker setup? Will docker see local changes?

Reply:
for me that's the biggest point of having the docker setup! docker-compose is set to mount the local directory so I can use it for development along with a database I can wipe at will.

Reply:
That's really interesting. I've only ever used Docker for deployment, not development.